File: app/Studifutter/MenuTableViewCell.m
//
//  MenuTableViewCell.m
//  Studifutter
//
//  Created by Florian Heiber on 23.11.11.
//  Copyright (c) 2011 rootof.net Heiber & Wiewel GbR. All rights reserved.
//

#import "MenuTableViewCell.h"
#import "Menu.h"

@interface MenuTableViewCell ()

- (void)updateLabels;

@end

@implementation MenuTableViewCell

@synthesize menu = _menu;

- (void)setMenu:(Menu *)menu {
    _menu = menu;
    [self updateLabels];
}

- (void)updateLabels {
    [nameLabel setText:[[self menu] name]];
    
    NSLocale *german = [[NSLocale alloc] initWithLocaleIdentifier:@"de_DE"];
    NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
    [formatter setLocale:german];
    [formatter setNumberStyle:NSNumberFormatterCurrencyStyle];
    
    NSString *priceString1 = [formatter stringFromNumber:[_menu price]];
    NSString *priceString2 = [formatter stringFromNumber:[_menu reducedPrice]];
    
    NSMutableString *infoString = [NSMutableString stringWithFormat:@"%@/%@", priceString1, priceString2];
    
    if (_menu.extraChars.length > 0) {
        [infoString appendFormat:@" - %@", _menu.extraChars];
    }
    
    if (_menu.extraNumbers.length > 0) {
        [infoString appendFormat:@" - %@", _menu.extraNumbers];
    }
    
    [infoLabel setText:infoString];
}

@end

File: app/Studifutter/MenuTableViewCell.swift
//
//  MenuTableViewCell.swift
//  Studifutter
//
//  Created by Florian Heiber on 14.10.2016.
//  Copyright © 2016 rootof.net Heiber & Wiewel GbR. All rights reserved.
//

import UIKit

class MenuTableViewCell: UITableViewCell {
    @IBOutlet weak var nameLabel: UILabel!
    @IBOutlet weak var infoLabel: UILabel!
    
    var meal: Meal?
    
    func configure(meal: Meal) {
        self.meal = meal
        
        nameLabel.text = meal.title
        
        let germanLocale = Locale(identifier: "de_DE")
        let numberFormatter = NumberFormatter()
        numberFormatter.locale = germanLocale
        numberFormatter.numberStyle = .currency
        
        // TODO: Generate all the prices here and create a string
        
        // TODO: Concatenate types into a nice string
        
    }
    
    override func prepareForReuse() {
        meal = nil
    }
}
